dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/TransferFunctionFreeze.mo

```diff
@@ -0,0 +1,94 @@
+within Dynawo.NonElectrical.Blocks.Continuous;
+
+/*
+* Copyright (c) 2024, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+block TransferFunctionFreeze "Linear transfer function"
+  extends Modelica.Blocks.Interfaces.SISO(y(start = y_start));
+   
+    //Input Parameter
+  Modelica.Blocks.Interfaces.BooleanInput freeze if UseFreeze annotation(
+    Placement(transformation(origin = {-70, 120}, extent = {{-20, -20}, {20, 20}}, rotation = -90), iconTransformation(origin = {-56, 120}, extent = {{-20, -20}, {20, 20}}, rotation = -90)));
+  
+  parameter Real b[:] = {1} "Numerator coefficients of transfer function (e.g., 2*s+3 is specified as {2,3})";
+  parameter Real a[:] = {1} "Denominator coefficients of transfer function (e.g., 5*s+6 is specified as {5,6})";
+  output Real x[nx](start = x_start) "State of transfer function from controller canonical form";
+  parameter Real x_start[nx] = zeros(nx) "Initial or guess values of states" annotation(
+    Dialog(group = "Initialization"));
+  parameter Real y_start = 0 "Initial value of output (derivatives of y are zero up to nx-1-th derivative)" annotation(
+    Dialog(group = "Initialization"));
+   parameter Boolean UseFreeze = false "= if true, freeze port enabled" annotation(
+  Evaluate = true,
+  HideResult = true,
+  choices(checkBox = true));
+  
+  protected
+  parameter Integer na = size(a, 1) "Size of denominator of transfer function";
+  parameter Integer nb = size(b, 1) "Size of numerator of transfer function";
+  parameter Integer nx = size(a, 1) - 1 "Size of vector x";
+  parameter Real bb[:] = vector([zeros(max(0, na - nb), 1); b]) "Numerator coefficients, padded if necessary by zeroes for highest-order coefficients";
+  parameter Real d = bb[1]/a[1] "Ratio of highest-order coefficients of numerator and denominator";
+  parameter Real a_end = if a[end] > 100*Modelica.Constants.eps*sqrt(a*a) then a[end] else 1 "Non-zero value of lowest-order coefficient of denominator";
+ 
+  Real x_scaled[nx](start = a_end*x_start) "Scaled vector x";
+IntegratorSetFreeze integrator(Y0=y_start,UseFreeze=UseFreeze);
+
+equation
+  assert(size(b, 1) <= size(a, 1), "Transfer function is not proper");
+    if nx == 0 then
+    integrator.u = der(d*u);
+  else
+    der(x_scaled[1])    = (-a[2:na]*x_scaled + a_end*u) / a[1];
+    der(x_scaled[2:nx]) = x_scaled[1:nx-1];
+    integrator.u = der(((bb[2:na] - d*a[2:na])*x_scaled) / a_end + d*u);
+    x = x_scaled / a_end;
+  end if;
+  if UseFreeze then
+  connect(freeze,integrator.freeze);
+  end if;
+  connect(integrator.y,y);
```

Comment:
You have replaced TransferFunction.y by der(TransferFunction.y), which is equal to integrator.u.

There is no need for a new block : TransferFunction followed by Derivative followed by IntegratorSetFreeze would be equivalent.

By the way, a Derivative followed by an IntegratorSetFreeze could be replaced by a structure freezing the output if a signal is true and leaving it unchanged if the signal is false, I will find such a diagram.